### PR TITLE
Add a flush method to support Lambda use cases

### PR DESCRIPTION
**Pull request by tredman**
All the context is here: https://github.com/honeycombio/libhoney-go/pull/26

^ this, but for python.

### CODE REVIEW

File: libhoney/__init__.py [hunk 1]
```diff
@@ -152,6 +152,16 @@ def send_now(data):
     ev.add(data)
     ev.send()
 
+def flush():
+    '''Closes and restarts the transmission, sending all events. Use this
+    if you want to perform a blocking send of all events in your
+    application.
+
+    Note: does not work with asyncronous Transmission implementations such
```
Comment (samstokes) [inline libhoney/__init__.py]: Nitpick: asynchronous

File: libhoney/client.py [hunk 2]
```diff
@@ -173,6 +173,19 @@ def close(self):
         # we should error on post-close sends
         self.xmit = None
 
+
+    def flush(self):
+        '''Closes and restarts the transmission, sending all events. Use this
+        if you want to perform a blocking send of all events in your
+        application.
+
+        Note: does not work with asyncronous Transmission implementations such
```
Comment (samstokes) [inline libhoney/client.py]: Ditto nitpick.

File: libhoney/client.py [hunk 3]
```diff
@@ -173,6 +173,19 @@ def close(self):
...
+        as TornadoTransmission.
+        '''
+        if self.xmit and isinstance(self.xmit, Transmission):
```
Comment (samstokes) [inline libhoney/client.py]: I'm not in love with this `isinstance` check (I'm not sure of the Pythonic way, but if this was Ruby I'd probably prefer `if self.xmit.respond_to?(:flush)`), but it seems fine while we only have two Transmission implementations (plus the dynamically mocked ones in the tests).

**Comment by tredman** [comment]
Lambda freezes the function instance after the handler returns - you are guaranteed execution of code inside the handler, but threads, for instance, are not guaranteed to run. The timeline may look something like this

0ms: function starts
20ms: libhoney.Event.send() called
30ms: function exits - function frozen
100ms: batch send thread runs <--- this never gets to run

For a busy function, the timeline can look something like:

0ms: function invocation 1
20ms: libhoney.Event.send() called
30ms: function exits - function frozen
40ms: function invocation 2
50ms: libhoney.Event.send() called
70ms: function exits - function frozen
80ms: function invocation 3
100ms: batch send thread runs, queue empty
105ms: libhoney.Event.send() called <-- never gets sent
120ms: function exits - function frozen

You can of course tweak the batch send interval but that doesn't really solve the problem, so the flush is needed to ensure the queue is sent before returning.

File: libhoney/client.py (hunk 3 above)
Comment (tredman) [reply libhoney/client.py]: We could just have a no-op implementation in the other Transmission - might have less overhead than inspecting the object on every call.

**Pull request merged**